findoutlie/outfind.py

```diff
@@ -3,13 +3,108 @@
 
 import os.path as op
 from glob import glob
+from functools import partial
 
+import numpy as np
+import nibabel as nib
 
-def detect_outliers(fname):
-    return [42]
+from findoutlie.detectors import std_detector, iqr_detector
+from findoutlie.metrics import dvars
+from findoutlie.spm_funcs import get_spm_globals
+
+def detect_outliers(fname, metric='vol_means', detector='std', 
```

Comment:
I like what you did with the combination of metrics here.  I'm wondering whether you could find some way of passing metrics with their corresponding options, because it's a bit clunky passing options specific to each metric as keyword arguments.

Options I can think of are:

* Passing the function as the metric, where you use the `partial` trick you see above to make a version of the function that prespecifies the options you want.
* Using the [**kwargs](https://www.pythontutorial.net/python-basics/python-kwargs/) trick to pass any keyword arguments, pass `**kwargs` to each metric function and then have each function select the kwargs that are relevant.